### DISCUSSION TITLE
How do I stitch vertical screenshots?

### DISCUSSION THREAD

**Discussion by AutonomousCat** [OpenStitching/stitching#182] [Q&A]
I have a lot of screenshots from a chat window and I'd like to stitch them together. How can I do this?

**Comment by lukasalexanderweber** [comment]
What have you tried? For Screenshots you need affine stitching.

**Reply by AutonomousCat** [reply]
Haven't tried anything because my use case doesn't seem to be documented. I thought maybe it's not possible so I thought I'd ask first. Does it automatically handle whether to stitch vertically or horizontally? My final product is expected to be around 600 x 15000px

**Reply by lukasalexanderweber** [reply]
[here](https://github.com/OpenStitching/stitching_tutorial/blob/master/docs/Stitching%20Tutorial.md#affine-stitcher) is the tutorial part that should also work for screenshots. Affine stitching should automatically handle it. However, there is a reported issue: https://github.com/OpenStitching/stitching/issues/125

**Reply by AutonomousCat** [reply]
I'm still confused how this works outside of a notebook. Does stitcher.stitch(..) return a Bytes like object? I know how to save an image manually but confused what the library actually does without any type hinting.

However right now I'm unable to run any stitching due to memory issue:
`Failed to allocate 147220070400 bytes in function 'cv::OutOfMemoryError'`

and then I tried lowering the "final_megapix" setting to 0.2 (tried 1 first and it gave the above error still) and it gave me
`cv2.error: OpenCV(4.8.1) D:\a\opencv-python\opencv-python\opencv\modules\core\src\ocl.cpp:5954: error: (-220:Unknown error code -220) OpenCL error CL_MEM_OBJECT_ALLOCATION_FAILURE (-4) during carror: (-220:Unknown error code -220) OpenCL error CL_MEM_OBJECT_All: clEnqueueReadBuffer(q, handle=00000207EB863FC0, CL_TRUE, 0, sz=29727129600, data=0000020EF7E5z=29727129600, data=0000020EF7E5C080, 0, 0, 0) in function 'cv::oC080, 0, 0, 0) in function 'cv::ocl::OpenCLAllocator::map'`

**Reply by lukasalexanderweber** [reply]
the output is a numpy array which can be saved using OpenCVs [imwrite()](https://docs.opencv.org/4.x/d4/da8/group__imgcodecs.html#gabbc7ef1aa2edfaa87772f1202d67e0ce).

To get a feeling with the package I would recomend downsizing the input images. You can use verbose stitching to get a feeling how your images are processed.
